edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MqttProtocolHead.cs

```diff
@@ -107,17 +108,16 @@ public async Task CloseAsync(CancellationToken token)
             {
                 this.logger.LogInformation("Stopping MQTT protocol head");
 
-                await (this.serverChannel?.CloseAsync() ?? TaskEx.Done);
-                await (this.eventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                await (this.parentEventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                await (this.wsEventLoopGroup?.ShutdownGracefullyAsync() ?? TaskEx.Done);
-                // TODO: gracefully shutdown the MultithreadEventLoopGroup in MqttWebSocketListener?
-                // TODO: this.webSocketListenerRegistry.TryUnregister("mqtts")?
+                await (this.serverChannel?.CloseAsync() ?? TaskEx.Done);
+                await TaskEx.TimeoutAfter(this.eventLoopGroup?.ShutdownGracefullyAsync(), TimeSpan.FromSeconds(TimeoutInSecs));

```

Comment:
nit: if you add using: using Microsoft.Azure.Devices.Edge.Util;

you can then have: this.eventLoopGroup?.ShutdownGracefullyAsync()..TimeoutAfter(timeout);

also timeout can be declared as TimeSpan, like static readonly TimeSpan TimeoutInSecs = TimeSpan.FromSeconds(30);